api/v1/installation_types.go

```diff
@@ -248,8 +248,22 @@ const (
 	BGPDisabled BGPOption = "Disabled"
 )
 
+type LinuxDataplaneOption string
+
+const (
+	LinuxDataplaneIptables LinuxDataplaneOption = "Iptables"
+	LinuxDataplaneBPF      LinuxDataplaneOption = "BPF"
+)
+
 // CalicoNetworkSpec specifies configuration options for Calico provided pod networking.
 type CalicoNetworkSpec struct {
+	// LinuxDataplane is used to select the dataplane used for Linux nodes. In particular, it
+	// causes the operator to add required mounts and environment variables for the particular dataplane.
+	// If not specified, iptables mode is used.
```

Comment:
We should default this field in `fillDefaults` in the `pkg/controller/installation/core_controller.go`, when CalicoNetwork is exists.